internal/core/lib.rs

```diff
@@ -172,3 +172,33 @@ pub fn open_url(url: &str) {
         debug_log!("Error opening url {}: {}", url, err);
     }
 }
+
+pub fn escape_markdown(text: &str) -> alloc::string::String {
```

Comment:
Note that this depends on where in markdown this goes.
If you have something like: 
````md
```
{}
```
````

then you need another way to escape.

I wonder if that kind of escaping make sense at all, and maybe it would be best to do the interpolation on our internal data structure after the parsing.

Reply:
What I'm suggesting is basically like:

```
let mut in_code = false;
for char in markdown {
   if char == '`' {
      in_code = !in_code;
   }
}
```

this works for both inline code and code blocks due to code blocks using 3 backticks.